charmcraft/commands/store/registry.py

```diff
@@ -64,13 +66,25 @@ def assert_response_ok(response, expected_status=200):
 class OCIRegistry:
     """Interface to a generic OCI Registry."""
 
-    def __init__(self, server, organization, image_name):
+    def __init__(self, server, image_name, *, username="", password=""):
         self.server = server
-        self.orga = organization
-        self.name = image_name
-
+        self.image_name = image_name
         self.auth_token = None
-        self.auth_encoded_credentials = None
+
+        if username:
+            _u_p = "{}:{}".format(username, password)
```

Comment:
this assumes password is set but not checked for, what are the implecations of setting "<username>:" ?

Reply:
This makes no assumptions on password. If the username is set, this means there are credentials, so `auth_encoded_credentials` is calculed, otherwise is set to None (later will be used by the `_authenticate` method.

If password is "foo", "asjkldhakl", or "", it's just fine.